src/basic/fs-util.h

```diff
@@ -132,6 +132,12 @@ int open_mkdir_at(int dirfd, const char *path, int flags, mode_t mode);
 
 int openat_report_new(int dirfd, const char *pathname, int flags, mode_t mode, bool *ret_newly_created);
 
+/* We define extended open flags from bits 33-64 to avoid conflicts with the existing open flags which could
+ * be assigned bits 1-32. */
```

Comment:
can we add a static assert to ensure this never happens?

Reply:
What did you have in mind? There's no mask for open flags that we can use here.

Reply:
There isn't, but maybe use the biggest flag that is known at this time? There aren't that many flags added to `open`, so it should not be difficult to keep up. What I'm worried about is clashes on weird architectures that would go unnoticed because nobody here builds or runs with them.

Reply:
syscall signatures are always the same no? Is there any architecture where openat() does not take `int` flags?

Reply:
Sure, assuming every architecture is sane and does that plus always has the same size, it all works. Is everything sane out there? :-) What I'm saying is that an assertion is very cheap, and gives a little bit of peace of mind at no cost